src/components/visualization/vegaLiteViz/vegaLiteViz.tsx

```diff
@@ -2,35 +2,53 @@ import './vegaLiteViz.css'
 
 import type { Theme } from '@mui/material/styles'
 import Typography from '@mui/material/Typography'
+import Box from '@mui/system/Box'
 import Stack from '@mui/system/Stack'
 import useTheme from '@mui/system/useTheme'
 import React, { useEffect, useRef, useState } from 'react'
 import { default as VegaEmbed } from 'vega-embed'
 
+import PopoverMenu from '../../menu/popoverMenu'
 import type { VegaLiteData } from '../../types'
 
+interface downloadUrls {
+  svg?: string
+  png?: string
+}
+
 /** The `VegaLiteViz` component is a versatile tool for visualizing data using the Vega-Lite grammar. With support for various graphic types, it empowers users to create engaging and informative data visualizations effortlessly. */
 function VegaLiteViz(props: VegaLiteData) {
   const chartRef = useRef<HTMLDivElement>(null)
   const [hasError, setHasError] = useState<boolean>(false)
   const rusticTheme: Theme = useTheme()
   const isDarkTheme = rusticTheme.palette.mode === 'dark'
   const defaultFont = rusticTheme.typography.fontFamily
+  const [downloadUrls, setDownloadUrls] = useState<downloadUrls>()
 
   function renderChart() {
     if (chartRef.current && props.spec) {
       const options = {
         config: { font: defaultFont },
         ...props.options,
         theme: isDarkTheme ? props.theme?.dark : props.theme?.light,
+        actions: false,
       }
 
       if (!props.options?.config?.font) {
         options.config.font = defaultFont
       }
 
       VegaEmbed(chartRef.current, props.spec, options)
-        .then(() => {
+        .then((result) => {
+          const urls: downloadUrls = {}
+          result.view.toImageURL('svg').then((url) => {
```

Comment:
I think it should be used with scaleFactor for consistency. The file name should also be similar to how its done in vega-embed using options. We could use `result.embedOptions` to get the required details.
Refer - https://github.com/vega/vega-embed/blob/c3b6c19bad7054e6a5ec0f3ca2647a888b4aac00/src/embed.ts#L478